tests/platform_tests/cli/test_show_platform_dpu.py

```diff
@@ -0,0 +1,183 @@
+    # Checking the state of dpu health again after bringing down interface and deleting db entry
+    dpu_db = wait_until(60, 60, 0, count_dpu_modules_in_system_health_cli, duthost)
+
+    pytest_assert(dpu_db == 0, "Link is not down'{}'".format(duthost.hostname))
+
+    output_interface_cmd = duthost.show_and_parse('show ip interface')
+    for index in range(len(output_interface_cmd)):
+        parse_output = output_interface_cmd[index]
+        interface = parse_output['interface']
+        status = parse_output['admin/oper']
+
+        if interface != "eth0" and 'eth' in interface:
+            if status == "down/down":
+                count_down += 1
+            output_intf_up = duthost.shell("ifconfig %s up"%(interface))["stdout_lines"]
+            time.sleep(1)
+
+    output_interface_cmd = duthost.show_and_parse('show ip interface')
+    for index in range(len(output_interface_cmd)):
+        parse_output = output_interface_cmd[index]
+        interface = parse_output['interface']
+        status = parse_output['admin/oper']
+
+        if interface != "eth0" and 'eth' in interface:
+            if status == "up/up":
+                count_up += 1
+
+    # Checking the state of dpu health again after bringing up interface and waiting for db entry to get populated
+    dpu_db_after = wait_until(60, 60, 0, count_dpu_modules_in_system_health_cli, duthost)
+
+    pytest_assert(dpu_db_before == count_up == count_down == dpu_db_after, "Link Flap is Tested'{}'".format(duthost.hostname))
+
+def test_shutdown_power_up_dpu(duthosts, enum_rand_one_per_hwsku_hostname, platform_api_conn):
+    """
+    @summary: Verify `shut down and power up DPU`
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    num_modules = int(chassis.get_num_modules(platform_api_conn))
+
+    for index in range(num_modules):
+        dpu = module.get_name(platform_api_conn, index)
+        duthosts.shell("config chassis modules shutdown %s"%dpu)
+        time.sleep(2)
+
+    pytest_assert(wait_until(120, 60, 0, check_dpu_module_status, duthost, num_modules, "off"),
+                          "Not all DPUs operationally down")
+
+    for index in range(num_modules):
+        dpu = module.get_name(platform_api_conn, index)
+        duthosts.shell("config chassis modules startup %s"%dpu)
+        time.sleep(2)
+
+    pytest_assert(wait_until(120, 60, 0, check_dpu_module_status, duthost, num_modules, "on"),
+                          "Not all DPUs operationally up")
+
+def test_reboot_cause(duthosts, enum_rand_one_per_hwsku_hostname, platform_api_conn):
+    """
+    @summary: Verify `Reboot Cause`
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    num_modules = int(chassis.get_num_modules(platform_api_conn))
+
+    for index in range(num_modules):
+        dpu_name = module.get_name(platform_api_conn, index)
+        output_shutdown_dpu = duthost.shell("config chassis module shutdown %s"%(dpu_name))["stdout_lines"]
+        time.sleep(2)
```

Comment:
will this sleep being enough for waiting for the shutdown? should we use other more stable signal?

Reply:
Changed it to wait until DPUs turns off.